pages/user/[username]/index.tsx

```diff
@@ -82,3 +83,35 @@ const Contributor: WithPageLayout = () => {
 
 Contributor.PageLayout = ProfileLayout;
 export default Contributor;
+
+export const getServerSideProps = async (context: UserSSRPropsContext) => {
+  return await handleUserSSR(context);
+};
+
+export type UserSSRPropsContext = GetServerSidePropsContext<{ username: string }>;
+
+export async function handleUserSSR({ params }: GetServerSidePropsContext<{ username: string }>) {
+  const { username } = params!;
+  const sessionResponse = await supabase.auth.getSession();
+  const sessionToken = sessionResponse?.data.session?.access_token;
+
+  const req = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/users/${username}`, {
+    headers: {
+      accept: "application/json",
+      Authorization: `Bearer ${sessionToken}`
+    }
+  });
+
+  const socialCardUrl = `${String(process.env.NEXT_PUBLIC_OPENGRAPH_URL ?? "")}/users/${username}`;
+  let ogImage = socialCardUrl;
+  const ogReq = await fetch(`${socialCardUrl}/metadata`); //statsu returned: 204 or 304 or 404
```

Comment:
this is not correct, `x-amz-meta-location` will always return the correct url, this endpoint is so the user knows they have to load the backend URL to generate the image. What you are doing here is changing the OG image url instead of working around the server side mechanism

Reply:
@0-vortex Hope this new commit correctly addresses the issue